### ISSUE TITLE
bundle/compile produces invalid code for variable assignment with await

### ISSUE THREAD

**Issue by nktpro** [denoland/deno#9810]
Here's a reproducer:

```typescript
let bar;

async function foo() {
  return await (bar = Promise.resolve(2));
}

await foo();
```

`deno bundle` would produce this:

```typescript
let bar;
async function foo() {
    return await bar = Promise.resolve(2);
}
await foo();
```

Notice that the parentheses are stripped around the assignment of `bar = ...`, which is invalid (`SyntaxError: Invalid left hand side for assignment`)

I ran into this while bundling code depending on `Ajv`, which [has a snippet with this exact code pattern](https://github.com/ajv-validator/ajv/blob/db553ad00d8046ffc75b821c12a3e3ab5cea38d0/lib/core.ts#L407).

Also this seems to go way back to deno `1.6.x`, it's not a recent regression.

**Comment by kitsonk** [comment]
cc/ @kdy1

**Comment by nktpro** [comment]
Confirmed this is now fixed in deno `1.9.0` release. Thanks @kdy1, you totally rock!
